dependencyManagement/build.gradle.kts

```diff
@@ -35,6 +37,7 @@ rootProject.extra["caffeine3Version"] = "3.0.4"
 val DEPENDENCY_BOMS = listOf(
   "com.fasterxml.jackson:jackson-bom:2.12.3",
   "com.google.guava:guava-bom:30.1.1-jre",
+  "io.grpc:grpc-bom:${grpcVersion}",
```

Comment:
I wouldn't expect this to work well with the dependencies declared in our `grpc` instrumentation modules - can you verify with `:dependencies` and `:dependencies -PtestLatestDeps=true` that they are as expected? If not, we could just list the `grpc-netty-shaded` in `DEPENDENCIES` below

Reply:
One example of the needed tweaks is this

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/netty/netty-4.0/javaagent/build.gradle.kts#L57

We do use netty-bom, mostly to avoid having dozens of netty versions loaded into our build which hasn't been as problematic for grpc. It requires this sort of syntax to make sure the BOM doesn't cause the wrong dependency in the instrumentation.